enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/components/Notebook.tsx

```diff
@@ -38,11 +58,9 @@ export const Notebook = ({ onApply = () => {} }: NotebookProps) => {
           isResultDirty={isDirty}
           reportTimezone={reportTimezone}
           readOnly={false}
-          updateQuestion={nextQuestion =>
-            updateQuestion(nextQuestion, { run: false })
-          }
-          runQuestionQuery={() => {
-            runQuestion();
+          updateQuestion={handleUpdateQuestion}
+          runQuestionQuery={async () => {
+            await runQuestion();
```

Comment:
what difference does `await`ting `runQuestion` make?

Reply:
I was running into issues where `onApply` was running before `runQuestion` finished, and caused `question` to reset to an empty state

Reply:
that makes sense if `onApply` _really_ needs to run after `runQuestion` promise is done.